delete fs canonicalize for showing broken softlink without error

zwpaper opened the pull request:
fix https://github.com/Peltoche/lsd/issues/72

I have checked:
1.  it could report file not existed for not existed files without the `canonicalize`
2. deleting `canonicalize` not broken any tests

And if we add the `canonicalize`, https://github.com/Peltoche/lsd/issues/72 happened

zwpaper commented:
BTW, I found `print_error` not appending newline,

maybe this should also be fixed?

![image](https://user-images.githubusercontent.com/3764335/78472533-2a6a3c00-776c-11ea-867b-121dc0648d20.png)

meain commented:
@zwpaper Yeah, I think it would be better for the error to have a newline.

zwpaper commented:
@meain I was trying to fix the `print_error`, but macro is beyond my ability, 
could you give me some advice on the improvement?

```
            let res = handle.write_all(std::format!($($arg)*).as_bytes());
```

this printed without newline, how could I fix this?

meain commented:
@zwpaper I would go about just adding a `\n` in `print_error` call. That would be the better way in my opinion. Because this macro could be reused even when we don't need `\n`. 
```rust
print_error!("cannot access '{}': {}\n", path.display(), err);
```

----

Just so that you know, all that is happening in that is that we are expanding everything that was inside `print_error!` call into `std.format`. So if we had to change and add a `\n` we will have to either capture the first arg separately and add `\n` to it. Another option would be to have the `std.format` captured in a variable and use another `std.format` to append `\n` to it.

zwpaper commented:
@meain I added a `newline` for each print_error, there should be a `newline` for the most cases, If there is any duplicated found later, I will fix it.
And I have changed the error message to match the `ls` command like
![image](https://user-images.githubusercontent.com/3764335/78634886-8dc4ad00-78d7-11ea-97be-da14c07b5617.png)

---
BTW, Maybe an icon for the error message should be more `lsd style`?
I will fire another PR and discuss for it.

meain commented:
@zwpaper I personally don't think `lsd: ` is a good idea. It does not add anything useful. I think we will stick with the current format or at least `lsd: cannot access '{}': ...`

Looks like there is not much consistency in how anyone does this.

![](https://i.imgur.com/NgQndi3.png)


One thing I did notice is that `lsd` is the only one that expands the symlink without `-l`. Maybe we should change that behavior.

zwpaper commented:
IMO, the error message explained the error, making the `cannot access` a little bit duplicated.
just like there is only one of the other three commands printing the `cannot access`

agree with the symlink expanding part.

meain commented:
Hmm, I guess that makes sense. We will stick to `lsd: ` then.

meain commented:
Looks like travis messed up something, I have restarted the build. Once that passes I will merge the PR.
